### PR TITLE
Accounting Categories API

### PR DESCRIPTION
**Pull request by zizouet**
Allows the user to see items categories and will allow him in the futur to filter his budget and balance

- [x] CategoriesAPI
- [x] SubCategoriesAPI
- [x] Documentation
- [x] Tests

**+ other dependencies updated from the main made by Sarah. As our PRs were dependant the only choice was to make Sarah's PR push on my branch. See PR #268 for Sarah's independent code which was reviewed separately. This PR will review François's code which has approx 400 lines**

### CODE REVIEW

File: app/src/main/java/com/github/se/assocify/model/database/AccountingCategoryAPI.kt [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+  /**
+   * Get the categories of an association
+   *
+   * @param associationUID the unique identifier of the association
+   * @param onSuccess the callback to be called when the categories are retrieved
+   * @param onFailure the callback to be called when the categories could not be retrieved
+   * @return the list of categories
+   */
+  fun getCategories(
```
Comment (ortoLover) [inline app/src/main/java/com/github/se/assocify/model/database/AccountingCategoryAPI.kt:21]: The get does not get the values from the database and throws an error. You should check on that

**Review by ortoLover — CHANGES_REQUESTED** [review]
There is a major problem: on the app it tells me that there are account subcategories even if there are some present in the repository. I did some checks and it seems like the getcategories throws an error, you might check onto that. But apart from that everything seems to work as expected, good job!!

File: app/src/main/java/com/github/se/assocify/model/entities/AccountingCategory.kt [hunk 2]
```diff
@@ -3,9 +3,10 @@ package com.github.se.assocify.model.entities
 /**
  * Represents the category of an accounting item.
  *
+ * @param uid uid of the category
  * @param name name of the category (ex: Pole)
  */
 
 // admin or presidence should be able to add new categories or modify it
```
[the remark this answers is not in the document]
Comment (zizouet) [reply app/src/main/java/com/github/se/assocify/model/entities/AccountingCategory.kt:10]: Comes from @sarahbadr17 PR it means that only admin or presidence should be able to add new categories or modify it

**Review by ortoLover — APPROVED** [review]
Everything is ok now. For the major problem, it turns out I was wrong and for a strange reason my emulator sent an error without any reason. But overall it is a good PR and will help a lot for the future of the app!

**Pull request merged**